src/main/java/com/microsoft/sqlserver/jdbc/SQLServerBlob.java

```diff
@@ -142,7 +142,20 @@ private void checkClosed() throws SQLServerException {
     public InputStream getBinaryStream() throws SQLException {
         checkClosed();
 
-        return getBinaryStreamInternal(0, value.length);
+        if( null == value)
+        {           
+            InputStream stream = (InputStream) activeStreams.get(0);
```

Comment:
I see this several times and this looks extremely brittle. I would consider making a dedicated field for this.